format not a string literal and no format arguments [-Werror=format-security]

Building DDNet 19.1 fails with


```
[86/318] Building CXX object CMakeFiles/engine-shared.dir/src/base/system.cpp.o
FAILED: CMakeFiles/engine-shared.dir/src/base/system.cpp.o 
/usr/bin/c++ -DCONF_DEBUG -DCONF_DISCORD -DCONF_DISCORD_DYNAMIC -DCONF_INFORM_UPDATE -DCONF_MYSQL -DCONF_OPENSSL -DCONF_UPNP -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"19.1\" -DGLEW_STATIC -D_FILE_OFFSET_BITS=64 -I/build/ddnet/src/build/src -I/build/ddnet/src/ddnet-19.1/src -I/build/ddnet/src/ddnet-19.1/src/rust-bridge -isystem /usr/include/miniupnpc -isystem /usr/include/mysql -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/ddnet/src=/usr/src/debug/ddnet -flto=auto -ffat-lto-objects -g -std=c++17 -fdiagnostics-color=always -fstack-protector-strong -fno-exceptions -fsigned-char -Wall -Wextra -Wno-psabi -Wno-unused-parameter -Wno-missing-field-initializers -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=global -Wsuggest-override -Wclass-memaccess -MD -MT CMakeFiles/engine-shared.dir/src/base/system.cpp.o -MF CMakeFiles/engine-shared.dir/src/base/system.cpp.o.d -o CMakeFiles/engine-shared.dir/src/base/system.cpp.o -c /build/ddnet/src/ddnet-19.1/src/base/system.cpp
In file included from /build/ddnet/src/ddnet-19.1/src/base/system.cpp:19:
/build/ddnet/src/ddnet-19.1/src/base/system.cpp: In function ‘void net_addr_str(const NETADDR*, char*, int, bool)’:
/build/ddnet/src/ddnet-19.1/src/base/system.cpp:1171:35: error: format not a string literal and no format arguments [-Werror=format-security]
 1171 |                 dbg_assert(false, error);
      |                                   ^~~~~
/build/ddnet/src/ddnet-19.1/src/base/system.h:87:60: note: in definition of macro ‘dbg_assert’
   87 |                         dbg_assert_imp(__FILE__, __LINE__, fmt, ##__VA_ARGS__); \
      |                                                            ^~~
At top level:
cc1plus: note: unrecognized command-line option ‘-Wno-nullability-completeness’ may have been intended to silence earlier diagnostics
cc1plus: some warnings being treated as errors
```

Happens with all buildtypes I tested: RelWithDebInfo, Release e Debug

Full build logs:
- [Fedora Rawhide ddnet package](https://github.com/user-attachments/files/19509782/Fedora-build.log) (see [build info page](https://koji.fedoraproject.org/koji/taskinfo?taskID=130845637) for more logs)
- [Arch Linux AUR ddnet package](https://github.com/user-attachments/files/19510037/ddnet-19.1-1-x86_64-build.log)

--- a/src/base/system.cpp
+++ b/src/base/system.cpp
@@ -1166,9 +1166,7 @@ void net_addr_str(const NETADDR *addr, char *string, int max_length, bool add_po
 	}
 	else
 	{
-		char error[64];
-		str_format(error, sizeof(error), "unknown NETADDR type %d", addr->type);
-		dbg_assert(false, error);
+		dbg_assert(false, "unknown NETADDR type %d", addr->type);
 	}
 }
 

--- a/src/engine/client/backend_sdl.cpp
+++ b/src/engine/client/backend_sdl.cpp
@@ -165,7 +165,7 @@ void CGraphicsBackend_Threaded::ProcessError(const SGfxErrorContainer &Error)
 		else
 			VerboseStr.append(ErrStr.m_Err);
 	}
-	dbg_assert(false, VerboseStr.c_str());
+	dbg_assert(false, "%s", VerboseStr.c_str());
 }
 
 bool CGraphicsBackend_Threaded::GetWarning(std::vector<std::string> &WarningStrings)

